engine/api/src/main/scala/pl/touk/nussknacker/engine/api/typed/typing.scala

```diff
@@ -125,13 +123,24 @@ object typing {
     def typedClass[T: ClassTag]: TypedClass = typedClass(toRuntime[T])
 
     //TODO: make it more safe??
-    def typedClass(klass: Class[_], parameters: List[TypingResult] = Nil): TypedClass = if (klass == classOf[Any]) {
-      throw new IllegalArgumentException("Cannot have typed class of Any, use Unknown")
-    } else if (klass.isPrimitive) {
-      TypedClass(ClassUtils.primitiveToWrapper(klass), parameters)
-    } else {
-      TypedClass(klass, parameters)
-    }
+    def typedClass(klass: Class[_], parameters: List[TypingResult] = Nil): TypedClass =
+      if (klass == classOf[Any]) {
+        throw new IllegalArgumentException("Cannot have typed class of Any, use Unknown")
+      } else if (klass.isPrimitive) {
+        TypedClass(ClassUtils.primitiveToWrapper(klass), parameters)
+      } else if (klass.isArray) {
+        //to not have separate class for each array, we pass Array of Objects
+        if (klass.getComponentType == classOf[Object]) {
+          TypedClass(klass, parameters)
+        } else parameters match {
+          case Nil =>
+            Typed.typedClass(classOf[Array[Object]], List(Typed(klass.getComponentType)))
+          case _: List[TypingResult] =>
+            throw new IllegalArgumentException("Array parameter type passed twice")
```

Comment:
We could add passed types (both from parameters and from componentType) in message?